ui/app/src/utils/content.ts

```diff
@@ -1092,3 +1093,22 @@ export function generateComponentBasePath(contentType: string) {
 export function generateComponentPath(modelId: string, contentType: string) {
   return `${generateComponentBasePath(contentType)}/${modelId}.xml`;
 }
+
+/**
+ * If the field is inherited, swaps the modelId and parentModelId with
+ * the inheritance parent's. */
+export function getInheritanceParentIdsForField(
+  fieldId: string,
+  modelLookup: LookupTable<ContentInstance>,
+  modelId: string,
+  parentModelId: string,
+  modelIdByPath: LookupTable<string>,
+  hierarchyMap: ModelHierarchyMap
+): { modelId: string; parentModelId: string } {
```

Comment:
@jvega190 I made some updates. You didn't add types and the function name was way too vague. Also the doc comment is always appreciated. Other minor changes on the other files; review changes and port to 4.2.